packages/api-gateway/src/services/validators.ts

```diff
@@ -93,3 +95,21 @@ export function assertRegistryAttributeExists(
     throw attributeNotFoundInRegistry(attributeId);
   }
 }
+
+export function assertIsEserviceProducerDelegate(
+  delegation: delegationApi.Delegation | undefined,
+  organizationId: TenantId
+): void {
+  if (!delegation || delegation.delegateId !== organizationId) {
+    throw operationForbidden;
+  }
+}
+
+export function assertOnlyOneActiveDelegationForEserviceExists(
+  delegations: delegationApi.Delegations,
+  eserviceId: apiGatewayApi.EService["id"]
+): void {
+  if (delegations.totalCount > 1) {
+    throw multipleActiveDelegationsForEservice(eserviceId);
+  }
+}
```

Comment:
In theory these two should keep into account also the delegation kind:
- There can be only one active delegation per kind: you can have a consumer and a producer delegation together
- The assertIsEserviceProducerDelegate shall also check the delegationKind

In practice, consumer delegations now exist in the model but cannot yet be created, so we could also add these checks as part of feature/incaricato, as you prefer

Reply:
Fixed in [PR](https://github.com/pagopa/interop-be-monorepo/pull/1239)